indico/modules/auth/providers.py

```diff
@@ -30,6 +33,34 @@ def check_password(self, identity, password):
             session['insecure_password_error'] = error
         return True
 
+    def process_local_login(self, data):
+        identity = self.identity_model.query.filter(type(self).provider_column == self.name,
```

Comment:
Remove the inheritance from `SQLAlchemyAuthProviderBase` and inherit from `AuthProviderBase` instead, and use the column/model names directly in here  instead of defining them on the class level (this was only useful when using the base class from flask-multipass).

Also, you can most likely write a single query that gets the local identities based on both the usual identifier match and email matches. That way you can avoid a large mount of code below.

Reply:
Cannot find `AuthProviderBase` is it `AuthProvider`?

If I use a query such as below
```
identity = (Identity.query.filter(Identity.provider == self.name,
                                          db.or_(Identity.identifier == data['identifier'],
                                                 UserEmail.email == data['identifier']),
                                          ~User.is_pending,
                                          ~UserEmail.is_user_deleted)
                                  .join(User)
                                  .join(UserEmail).first())
```
 Then I'll still need to do another query below to show that the user is logging in with an email address that isn't the primary email. Or we eliminate that part?

Reply:
I'd check `data['identifier'] in user.secondary_emails` for that

Reply:
Okay